protocols/discord.py

```diff
@@ -421,34 +467,46 @@ def on_message(self, event: MessageCreate, *args, **kwargs):
             subserver = message.guild.id
             target = message.channel.id
 
-        if subserver:
-            pylink_netobj = self.protocol._children[subserver]
-            def format_user_mentions(u):
-                # Try to find the user's guild nick, falling back to the user if that fails
-                if message.guild and u.id in message.guild.members:
-                    return '@' + message.guild.members[u.id].name
-                else:
-                    return '@' + str(u)
+        if not subserver:
+            return
+
+        def format_user_mentions(u):
+            # Try to find the user's guild nick, falling back to the user if that fails
+            if message.guild and u.id in message.guild.members:
+                return '@' + message.guild.members[u.id].name
+            else:
+                return '@' + str(u)
 
             # Translate mention IDs to their names
-            text = message.replace_mentions(user_replace=format_user_mentions,
-                                            role_replace=lambda r: '@' + str(r),
-                                            channel_replace=str)
-            try:
-                text = D2IFormatter().format(text)
-            except:
-                log.exception('(%s) Error translating from Discord to IRC: %s', self.name, text)
 
-            def _send(text):
-                for line in text.split('\n'):  # Relay multiline messages as such
-                    pylink_netobj.call_hooks([message.author.id, 'PRIVMSG', {'target': target, 'text': line}])
+        text = message.replace_mentions(user_replace=format_user_mentions,
+                                        role_replace=lambda r: '@' + str(r),
+                                        channel_replace=str)
+        try:
+            text = D2IFormatter().format(text)
+        except:
+            log.exception('(%s) Error translating from Discord to IRC: %s', self.name, text)
+
+        pylink_netobj = self.protocol._children[subserver]
+        author = message.author.id
+        if message.webhook_id:
+            if message.webhook_id == getattr(getattr(pylink_netobj.channels[target], 'webhook', None), 'id', None):
+                return
+
+            author = self.webhook_agent_uid
+            # Format the message to contain the webhook username
+            text = f'<{message.author.username}> {text}'
```

Comment:
My support target for now is Python 3.5, which doesn't have f-strings. Please do string substitution instead!